lib/rules/utils/ast-utils.js

```diff
@@ -2528,15 +2528,18 @@ module.exports = {
 	 */
 	areBracesNecessary(node, sourceCode) {
 		/**
-		 * Determines if the given node is a lexical declaration (let, const, function, or class)
+		 * Determines if the given node is a lexical declaration (let, const, using, await using, function, or class)
 		 * @param {ASTNode} nodeToCheck The node to check
 		 * @returns {boolean} True if the node is a lexical declaration
 		 * @private
 		 */
 		function isLexicalDeclaration(nodeToCheck) {
 			if (nodeToCheck.type === "VariableDeclaration") {
 				return (
-					nodeToCheck.kind === "const" || nodeToCheck.kind === "let"
+					nodeToCheck.kind === "const" ||
+					nodeToCheck.kind === "let" ||
+					nodeToCheck.kind === "using" ||
+					nodeToCheck.kind === "await using"
```

Comment:
Not a blocker but can we move these to a set like `const DECLARATION_KINDS = new Set(["const", "using", "await using"]);` and then assert on `DECLARATION_KINDS.has(nodeToCheck.kind)`